src/main/java/org/sonarlint/intellij/ui/nodes/SummaryNode.java

```diff
@@ -97,6 +99,28 @@ public int insertLiveSecurityHotspotNode(LiveSecurityHotspotNode newChild, Compa
     return insertIdx;
   }
 
+  public void updateLiveSecurityHotspotNodeFromFileNode(String securityHotspotKey, HotspotStatus status, FindingTreeIndex index) {
+    var fileNodes = children.stream().map(n -> (FileNode) n).collect(Collectors.<FileNode>toList());
+    for (var fileNode : fileNodes) {
+      var children = fileNode.children();
+      while(children.hasMoreElements()) {
+        var hotspotNode = (LiveSecurityHotspotNode) children.nextElement();
+        var hotspot = hotspotNode.getHotspot();
+        if (securityHotspotKey.equals(hotspot.getServerFindingKey())) {
+          hotspot.setStatus(status);
+          if (hotspot.isResolved()) {
+            fileNode.remove(hotspotNode);
+            if (fileNode.getFindingCount() == 0) {
+              index.remove(fileNode.file());
+              remove(fileNode);
```

Comment:
I am surprised that we need to look for the node by going down in the hierarchy and remove it ourselves, I would have expected other ways to achieve that using the model, but maybe I'm wrong

Reply:
Improved